models/repo.go

```diff
@@ -1903,6 +1905,128 @@ func (repos MirrorRepositoryList) LoadAttributes() error {
 	return repos.loadAttributes(x)
 }
 
+type ScrubSensativeDataOptions struct {
+	LastCommitID	string
+	CommitMessage	string
+}
+
+// SrubSenstaiveData removes emails and passwords from the manifest.json or project.json file's history.
+func (repo *Repository) ScrubSensativeData(doer *User, opts ScrubSensativeDataOptions)  error {
+	repoWorkingPool.CheckIn(com.ToStr(repo.ID))
+	defer repoWorkingPool.CheckOut(com.ToStr(repo.ID))
+
+	localPath := repo.LocalCopyPath()
+
+	if err := repo.DiscardLocalRepoBranchChanges("master"); err != nil {
+		return fmt.Errorf("DiscardLocalRepoBranchChanges [branch: master]: %v", err)
+	} else if err = repo.UpdateLocalCopyBranch("master"); err != nil {
+		return fmt.Errorf("UpdateLocalCopyBranch [branch: master]: %v", err)
+	}
+
+	dirty := false
+	dirty = repo.ScrubJsonFile("project.json") || dirty
+	dirty = repo.ScrubJsonFile("package.json") || dirty
+	dirty = repo.ScrubJsonFile("manifest.json") || dirty
+	dirty = repo.ScrubJsonFile("status.json") || dirty
```

Comment:
`settings.scrub_notices_1` says it will only scrub manifest.json and project.json